Sources/Subprocess/Platforms/Subprocess+Windows.swift

```diff
@@ -629,6 +750,148 @@ extension Executable {
+            }
+            // 1. The directory from which the application loaded.
+            let applicationDirectory = try? fillNullTerminatedWideStringBuffer(
+                initialSize: DWORD(MAX_PATH), maxSize: DWORD(MAX_PATH)
+            ) {
+                return GetModuleFileNameW(nil, $0.baseAddress, DWORD($0.count))
+            }
+            if let applicationDirectory {
+                insertExecutableAddingExtension(
+                    name,
+                    currentPath: applicationDirectory,
+                    pathExtensions: pathExtensions,
+                    storage: &possiblePaths
+                )
+            }
+            // 2. Current directory
+            let directorySize = GetCurrentDirectoryW(0, nil)
+            let currentDirectory = try? fillNullTerminatedWideStringBuffer(
+                initialSize: directorySize >= 0 ? directorySize : DWORD(MAX_PATH),
+                maxSize: DWORD(MAX_PATH)
+            ) {
+                return GetCurrentDirectoryW(DWORD($0.count), $0.baseAddress)
+            }
+            if let currentDirectory {
+                insertExecutableAddingExtension(
+                    name,
+                    currentPath: currentDirectory,
+                    pathExtensions: pathExtensions,
+                    storage: &possiblePaths
+                )
+            }
+            // 3. System directory (System32)
+            let systemDirectorySize = GetSystemDirectoryW(nil, 0)
+            let systemDirectory = try? fillNullTerminatedWideStringBuffer(
+                initialSize: systemDirectorySize >= 0 ? systemDirectorySize : DWORD(MAX_PATH),
+                maxSize: DWORD(MAX_PATH)
+            ) {
+                return GetSystemDirectoryW($0.baseAddress, DWORD($0.count))
+            }
+            if let systemDirectory {
+                insertExecutableAddingExtension(
+                    name,
+                    currentPath: systemDirectory,
+                    pathExtensions: pathExtensions,
+                    storage: &possiblePaths
+                )
+            }
+            // 4. 16 bit Systen Directory
+            // Windows documentation stats that
+            // "No such standard function (similar to GetSystemDirectory)
+            // exists for the 16-bit system folder". Use C:\Windows\System instead
+            let systemDirectory16 = FilePath(#"C:\Windows\System"#)
+            insertExecutableAddingExtension(
+                name,
+                currentPath: systemDirectory16.string,
+                pathExtensions: pathExtensions,
+                storage: &possiblePaths
+            )
+            // 5. The Windows directory
+            let windowsDirectorySize = GetSystemWindowsDirectoryW(nil, 0)
+            let windowsDirectory = try? fillNullTerminatedWideStringBuffer(
+                initialSize: windowsDirectorySize >= 0 ? windowsDirectorySize : DWORD(MAX_PATH),
+                maxSize: DWORD(MAX_PATH)
```

Comment:
maxSize needs to be `DWORD(Int16.max)` to correctly handle extended length paths. `MAX_PATH` is only 260.